### DISCUSSION TITLE
TreatLife DS02S Configuration

### DISCUSSION THREAD

**Discussion by boboxx** [arendst/Tasmota#15245] [Q&A] [answered]
### PROBLEM DESCRIPTION
I just loaded Tasmota 11.0 on a Treatlife DS02S. On initil load I could control the switch via the web interface but had no dimming. I followed the instruction using teh templae and extra command from (https://templates.blakadder.com/treatlife_DS02S.html) and now I can't even do a On/Off via web interface. The interface is changing but the bulb is not turning on or off. If i use the physical switch button everything is working. Anyone know what I could have done wrong?

Edit: I have also tried using teh new template with same resukts. {"NAME":"DS02S Dimmer","GPIO":[0,2272,0,2304,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":54} 

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
02:27:52.905 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 26, Data 'Template; Module; GPIO 255'
02:27:52.929 SRC: Backlog
02:27:52.931 CMD: Grp 0, Cmnd 'TEMPLATE', Idx 1, Len 0, Data ''
02:27:52.939 MQT: stat/tasmota_215B7C/RESULT = {"NAME":"DS02S Dimmer","GPIO":[0,2272,0,2304,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":54}
02:27:53.136 SRC: Backlog
02:27:53.138 CMD: Grp 0, Cmnd 'MODULE', Idx 1, Len 0, Data ''
02:27:53.144 MQT: stat/tasmota_215B7C/RESULT = {"Module":{"0":"DS02S Dimmer"}}
02:27:53.349 SRC: Backlog
02:27:53.352 CMD: Grp 0, Cmnd 'GPIO', Idx 1, Len 3, Data '255'
02:27:53.363 MQT: stat/tasmota_215B7C/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"2272":"Tuya Tx"},"GPIO2":{"0":"None"},"GPIO3":{"2304":"Tuya Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
02:27:54.414 TYA: Send "55aa00000000ff"

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
02:28:21.840 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 19, Data 'Rule1; Rule2; Rule3'
02:28:21.876 SRC: Backlog
02:28:21.879 CMD: Grp 0, Cmnd 'RULE', Idx 1, Len 0, Data ''
02:28:21.886 MQT: stat/tasmota_215B7C/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
02:28:22.084 SRC: Backlog
02:28:22.087 CMD: Grp 0, Cmnd 'RULE', Idx 2, Len 0, Data ''
02:28:22.098 MQT: stat/tasmota_215B7C/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
02:28:22.299 SRC: Backlog
02:28:22.301 CMD: Grp 0, Cmnd 'RULE', Idx 3, Len 0, Data ''
02:28:22.312 MQT: stat/tasmota_215B7C/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
02:28:27.419 TYA: Send "55aa00000000ff"
02:28:30.760 WIF: Checking connection...

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
02:28:53.511 CMD: Status 0
02:28:53.513 SRC: WebConsole from 10.1.11.11
02:28:53.516 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '0'
02:28:53.522 MQT: stat/tasmota_215B7C/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_215B7C","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
02:28:53.530 MQT: stat/tasmota_215B7C/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:18:46","StartupUTC":"2022-03-29T01:10:07","Sleep":50,"CfgHolder":4617,"BootCount":24,"BCResetTime":"2022-03-27T17:19:18","SaveCount":198,"SaveAddress":"F6000"}}
02:28:53.539 MQT: stat/tasmota_215B7C/STATUS2 = {"StatusFWR":{"Version":"11.0.0(tasmota)","BuildDateTime":"2022-02-12T14:13:50","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"343/699"}}
02:28:53.548 MQT: stat/tasmota_215B7C/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":4,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["SmartNet",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00908009","2805C80001000600003C5A0A190000000000","00000090","0000E000","00004000"]}}
02:28:53.564 MQT: stat/tasmota_215B7C/STATUS4 = {"StatusMEM":{"ProgramSize":620,"Free":380,"Heap":23,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
02:28:53.574 MQT: stat/tasmota_215B7C/STATUS5 = {"StatusNET":{"Hostname":"tasmota-215B7C-7036","IPAddress":"10.1.10.180","Gateway":"10.1.10.1","Subnetmask":"255.255.255.0","DNSServer1":"10.1.10.1","DNSServer2":"0.0.0.0","Mac":"24:A1:60:21:5B:7C","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
02:28:53.581 MQT: stat/tasmota_215B7C/STATUS6 = {"StatusMQT":{"MqttHost":"10.1.10.2","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_215B7C","MqttUser":"mqttuser","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
02:28:53.590 MQT: stat/tasmota_215B7C/STATUS7 = {"StatusTIM":{"UTC":"2022-03-29T01:28:53","Local":"2022-03-29T02:28:53","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:33","Sunset":"19:15"}}
02:28:53.599 MQT: stat/tasmota_215B7C/STATUS10 = {"StatusSNS":{"Time":"2022-03-29T02:28:53"}}
02:28:53.609 MQT: stat/tasmota_215B7C/STATUS11 = {"StatusSTS":{"Time":"2022-03-29T02:28:53","Uptime":"0T00:18:46","UptimeSec":1126,"Heap":23,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":1,"POWER":"ON","Dimmer":29,"Fade":"OFF","Speed":1,"LedTable":"OFF","Wifi":{"AP":1,"SSId":"SmartNet","BSSId":"DC:A5:F4:9D:85:12","Channel":1,"Mode":"11n","RSSI":100,"Signal":-50,"LinkCount":1,"Downtime":"0T00:00:03"}}}
02:29:00.416 TYA: Send "55aa00000000ff"

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
02:29:56.011 CMD: weblog 4
02:29:56.013 SRC: WebConsole from 10.1.11.11
02:29:56.016 CMD: Grp 0, Cmnd 'WEBLOG', Idx 1, Len 1, Data '4'
02:29:56.021 MQT: stat/tasmota_215B7C/RESULT = {"WebLog":4}
```

### TO REPRODUCE
_Steps to reproduce the behavior:_
 WebLog 4
 setoption97 1
 BackLog TuyaMCU 21,2
 SetOption20 1
 SetOption54 1
 Backlog TuyaSend2 3,100; DimmerRange 100,1000
 TuyaSend4 4,1
 Dimming Mode 1: SerialSend5 55AA00060005040400010013
 ledtable 0
 WebLog 2

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
control on/off and dimm

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
This is the fist Tim loading Tasmota

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by boboxx** [comment, marked as answer]
seeting back serial speed to 9600bps fix the issue (setoption97 0)

**Reply by rothnic** [reply]
Did you do anything else? I have multiple dimmers from the same box working, but my latest one I'm converting just won't ever respond to the `Send "55aa00000000ff"`. I tried changing the baud rate to each option and it doesn't help.

**Reply by boboxx** [reply]
What is the sticker version of your switch?

**Reply by rothnic** [reply]
Just opened a switch from the same box and the board looks identical to the one on [this page](https://templates.blakadder.com/treatlife_DS02S.html). The board version, date, chip product no, etc. There is a sticker on the switch with `XB-DS02S01V2`.
